src/main/java/io/anserini/qa/passage/WmdPassageScorer.java

```diff
@@ -0,0 +1,167 @@
+        stopWords.add(line);
+      }
+    }
+  }
+
+  public double distance(float[] leftVector, float[] rightVector) {
+    double sum = 0.0;
+
+    for (int i = 0; i < leftVector.length; i++) {
+      sum += Math.pow(leftVector[i] - rightVector[i], 2);
+    }
+    return  Math.sqrt(sum);
+  }
+
+  @Override
+  public void score(String query, Map<String, Float> sentences) throws Exception {
+    StandardAnalyzer sa = new StandardAnalyzer(StopFilter.makeStopSet(stopWords));
+    TokenStream tokenStream = sa.tokenStream("contents", new StringReader(query));
+    CharTermAttribute charTermAttribute = tokenStream.addAttribute(CharTermAttribute.class);
+    tokenStream.reset();
+
+    Set<String> questionTerms = new HashSet<>();
+    Set<String> candidateTerms = new HashSet<>();
+
+    // avoid duplicate passages
+    Set<String> seenSentences = new HashSet<>();
+
+    while (tokenStream.incrementToken()) {
+      questionTerms.add(charTermAttribute.toString());
+    }
+
+    for (Map.Entry<String, Float> sent : sentences.entrySet()) {
+      double wmd = 0.0;
+      candidateTerms.clear();
+      sa = new StandardAnalyzer(StopFilter.makeStopSet(stopWords));
+      TokenStream candTokenStream = sa.tokenStream("contents", new StringReader(sent.getKey()));
+      charTermAttribute = candTokenStream.addAttribute(CharTermAttribute.class);
+      candTokenStream.reset();
+
+      while (candTokenStream.incrementToken()) {
+        candidateTerms.add(charTermAttribute.toString());
+      }
+
+      for(String qTerm : questionTerms) {
+        double minWMD = Double.MAX_VALUE;
+        for (String candTerm : candidateTerms) {
+          try {
+            double thisWMD = distance(wmdDictionary.getEmbeddingVector(qTerm), wmdDictionary.getEmbeddingVector(candTerm));
+            if (minWMD > thisWMD) {
+              minWMD = thisWMD;
+            }
+          } catch (TermNotFoundException e) {
+            String missingTerm = e.getMessage();
+
+            //if the question term and the answer term both do not exist in the
+            //dictionary and question term equals to the answer term
+            if (qTerm.equals(missingTerm)) {
+              if (qTerm.equals(candTerm)) {
+                minWMD = 0.0;
+              } else {
+                try {
+                  //if the
+                  double thisWMD = distance(wmdDictionary.getEmbeddingVector("unk"), wmdDictionary.getEmbeddingVector(candTerm));
+                  if (minWMD > thisWMD) {
+                    minWMD = thisWMD;
+                  }
+                } catch (TermNotFoundException e1) {
+                  //"unk" is OOV
+                }
+              }
+            } else {
+              continue;
+            }
+          } catch (IOException e) {
+          }
+        }
+
+        if (minWMD != Double.MAX_VALUE) {
+          wmd += minWMD;
+        }
+      }
+
+//      // based on observation, ignore any sentences that have less than 4 tokens
```

Comment:
I think in general we should not have commented-out code. So if this is not needed then we can remove it instead of commenting it out? We can always retrieve it from the git history if need be.